src/Fmt_ast.ml

```diff
@@ -1424,8 +1424,7 @@ and fmt_expression c ?(box= true) ?epi ?eol ?parens ?ext
       let opn, cls =
         let can_skip_parens =
           match e0.pexp_desc with
-          | Pexp_array _ | Pexp_constraint _ | Pexp_record _ | Pexp_tuple _ ->
-              true
+          | Pexp_array _ | Pexp_constraint _ | Pexp_record _ -> true
```

Comment:
I'm not sure about this change, shouldn't the two modes behave the same in this case?

Reply:
With can only skip Pexp_tuple when parens_tuple = `Always